### PR TITLE
Bug fix: Missing tables in the assessment report recommendations

### PR DESCRIPTION
**Pull request by sanyamsinghal**
Thanks @makalaaneesh for help in figuring out the simplified queries.

Index count matching in Schema Summary and Sizing Reasoning
image

### CODE REVIEW

File: yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus [hunk 1]
```diff
@@ -14,21 +14,89 @@ WHENEVER OSERROR EXIT FAILURE
...
-    END AS table_name,
-    0 as row_count
-FROM
-    dba_segments s
-LEFT JOIN
-    dba_tab_subpartitions sp ON s.owner = sp.table_owner AND s.segment_name = sp.table_name AND s.partition_name = sp.subpartition_name
-WHERE
-    s.segment_type IN ('TABLE', 'TABLE PARTITION', 'TABLE SUBPARTITION')
-    AND s.owner = '&1';
+-- Inline view to get the base tables, partitions, and subpartitions with correct naming
```
Comment (makalaaneesh) [inline yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus:17]: @sanyamsinghal check this. I think this should do it...
```
SELECT DISTINCT 
    t.owner AS schema_name, 
    t.table_name,
    CASE 
      WHEN sp.subpartition_name IS NOT NULL THEN sp.table_name || '_' || sp.partition_name || '_' || sp.subpartition_name 
      WHEN  tp.partition_name IS NOT NULL THEN tp.table_name || '_' || tp.partition_name
      ELSE t.table_name 
    END AS table_name_generated,
    0 as row_count
  FROM
    dba_tables t
  LEFT JOIN 
    dba_tab_partitions tp ON t.owner = tp.table_owner AND t.table_name = tp.table_name
  LEFT JOIN
    dba_tab_subpartitions sp ON t.owner = sp.table_owner AND t.table_name = sp.table_name 
  WHERE 
    t.owner = 'SDAB';
```

**Comment by sanyamsinghal** [comment]
Jenkins Pipeline: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/2833/

File: yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus (hunk 1 above)
Comment (makalaaneesh) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus:17]: updated fyi

Comment (makalaaneesh) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus:17]:
```
SELECT  
    t.owner AS schema_name, 
    t.table_name,
    CASE 
      WHEN sp.subpartition_name IS NOT NULL THEN sp.table_name || '_' || sp.partition_name || '_' || sp.subpartition_name 
      WHEN  tp.partition_name IS NOT NULL THEN tp.table_name || '_' || tp.partition_name
      ELSE t.table_name 
    END AS table_name_generated,
    0 as row_count
  FROM
    dba_tables t
  LEFT JOIN 
    dba_tab_partitions tp ON t.owner = tp.table_owner AND t.table_name = tp.table_name
  LEFT JOIN
    dba_tab_subpartitions sp ON tp.table_owner = sp.table_owner AND tp.partition_name = sp.partition_name AND t.table_name = sp.table_name
  WHERE 
    t.owner = 'ASSESS_MIGRATION';
```

Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus]: done

**Pull request merged**